src/omotes_orchestrator/postgres_job_manager.py

```diff
@@ -0,0 +1,101 @@
+
+
+LOGGER = logging.getLogger("omotes_orchestrator")
+
+
+class PostgresJobManager:
+    """Periodically checks the job row in the database and cleans up the stale one."""
+
+    postgresql_if: PostgresInterface
+    """Interface to PostgreSQL."""
+    postgres_job_manager_config: PostgresJobManagerConfig
+    """PostgresJobManager configuration"""
+    _init_time: datetime
+    """Instantiated datetime (UTC), is typically instantiated when the orchestrator is started"""
+    _active_threshold_sec: int
+    """Only when the instance is initialized longer than the threshold period
+    can it start cleaning the stale jobs."""
+    _job_retention_sec: int
+    """The allowed retention time in seconds of a postgres job row"""
+    _rerun_sec: int
+    """The period in seconds to rerun the stale job cleaning task"""
+
+    def __init__(self,
+                 postgresql_if: PostgresInterface,
+                 postgres_job_manager_config: PostgresJobManagerConfig) -> None:
+        """Construct the postgres job manager."""
+        self.postgresql_if = postgresql_if
+        self.postgres_job_manager_config = postgres_job_manager_config
+        self._init_time = datetime.now(timezone.utc)
+        self._active_threshold_sec = self.postgres_job_manager_config.job_retention_sec
+        self._job_retention_sec = self.postgres_job_manager_config.job_retention_sec
+        self._rerun_sec = 30
+
+        self._stale_jobs_cleaner_thread = threading.Thread(target=self.stale_jobs_cleaner,
+                                                           daemon=True)
+        self._stale_jobs_cleaner_thread_active = False
+
+    def start(self) -> None:
+        """Start the postgres job manager activities as a daemon process."""
+        if not self._stale_jobs_cleaner_thread_active:
+            self._stale_jobs_cleaner_thread.start()
+            self._stale_jobs_cleaner_thread_active = True
+
+            LOGGER.info("Starting the postgres job manager")
+
+    def stop(self) -> None:
+        """Stop the postgres job manager activities."""
+        if self._stale_jobs_cleaner_thread_active:
+            self._stale_jobs_cleaner_thread.join(timeout=0.0)
+            self._stale_jobs_cleaner_thread_active = False
+
+            LOGGER.info("Stopped the postgres job manager")
+
+    def stale_jobs_cleaner(self) -> None:
+        """Start a background process to clean up stale jobs longer than the retention time.
+
+        The function periodically checks if there are any stale jobs/rows in the database
+        longer than the configured retention time. Meanwhile if the PostgresJobManager
+        is instantiated (usually when the orchestrator is started) longer than
+        the configured time, the job/row will be deleted outright.
+        """
+        while True:
+            cur_time = datetime.now(timezone.utc)
+            active_sec = (cur_time - self._init_time).total_seconds()
+
+            if active_sec > self._active_threshold_sec:
```

Comment:
Does this mean that the job manager will not start deleting job rows until it has been up more than 48h? An orchestrator restart will then affect the job cleanup process?  
Or should `self._active_thresholds_seconds` be set to a different value initially (see comment above)?

Reply:
Why does the orchestrator need to be up a specific time before the cleanup can happen?

Reply:
See https://github.com/Project-OMOTES/orchestrator/issues/38

Specifically:
`This component should safe (in memory) the datetime when the component was initialized. This time may be used to check how long the orchestrator has been active for. Only after the orchestrator has been online for the time period for which is checked (48 hours) should the orchestrator be confident the job/row is actually 48 hours without successful completion. Otherwise, the orchestrator may have been offline for 48 hours or longer and the rows would be removed erronuously.`

Reply:
In other words: Not only must the jobs be older than 48 hours, the orchestrator needs to be online and active to see for 48 hours they do not get resolved.